Port to dune 3.6

bikallem opened the pull request:
Fixes https://github.com/UnixJunkie/interval-tree/issues/6

bikallem committed:
port to dune 3.6

UnixJunkie commented:
Can you make the dune-project file more minimal and not require a bleeding-edge version of dune?

UnixJunkie commented:
thanks for taking a stab at this issue

UnixJunkie committed:
no thanks dune, don't generate the opam file

do one thing and do it well: just compile this project
the opam file is none of dune's business

UnixJunkie committed:
my standard Makefile wrapper for dune

UnixJunkie committed:
hand-written opam file

UnixJunkie committed:
indent opam file

UnixJunkie commented:
I eddited directly some of the files from your PR.
Thanks a lot for your contribution.
I will update the package in opam shortly.
